activerecord/test/cases/migration/compatibility_test.rb

```diff
@@ -269,81 +269,120 @@ def migrate(x)
+    assert connection.index_exists?(:testings, [:gizmo_type, :gizmo_id], name: :index_testings_on_gizmo_type_and_gizmo_id)
+  end
 
-        assert connection.index_exists?(:more_testings, [:widget_type, :widget_id], name: :index_more_testings_on_widget_type_and_widget_id)
-        assert connection.index_exists?(:more_testings, [:gizmo_type, :gizmo_id], name: :index_more_testings_on_gizmo_type_and_gizmo_id)
-      ensure
-        connection.drop_table :more_testings rescue nil
+  def test_create_join_table_with_polymorphic_reference_uses_all_column_names_in_index
+    migration = Class.new(migration_class) {
+      def migrate(x)
+        create_join_table :more, :testings do |t|
+          t.references :widget, polymorphic: true, index: true
+          t.belongs_to :gizmo, polymorphic: true, index: true
+        end
       end
+    }.new
 
-      def test_polymorphic_add_reference_uses_all_column_names_in_index
-        migration = Class.new(ActiveRecord::Migration[6.0]) {
-          def migrate(x)
-            add_reference :testings, :widget, polymorphic: true, index: true
-            add_belongs_to :testings, :gizmo, polymorphic: true, index: true
-          end
-        }.new
+    ActiveRecord::Migrator.new(:up, [migration], @schema_migration).migrate
 
-        ActiveRecord::Migrator.new(:up, [migration], @schema_migration).migrate
+    assert connection.index_exists?(:more_testings, [:widget_type, :widget_id], name: :index_more_testings_on_widget_type_and_widget_id)
+    assert connection.index_exists?(:more_testings, [:gizmo_type, :gizmo_id], name: :index_more_testings_on_gizmo_type_and_gizmo_id)
+  ensure
+    connection.drop_table :more_testings rescue nil
+  end
 
-        assert connection.index_exists?(:testings, [:widget_type, :widget_id], name: :index_testings_on_widget_type_and_widget_id)
-        assert connection.index_exists?(:testings, [:gizmo_type, :gizmo_id], name: :index_testings_on_gizmo_type_and_gizmo_id)
+  def test_polymorphic_add_reference_uses_all_column_names_in_index
+    migration = Class.new(migration_class) {
+      def migrate(x)
+        add_reference :testings, :widget, polymorphic: true, index: true
+        add_belongs_to :testings, :gizmo, polymorphic: true, index: true
+      end
+    }.new
+
+    ActiveRecord::Migrator.new(:up, [migration], @schema_migration).migrate
+
+    assert connection.index_exists?(:testings, [:widget_type, :widget_id], name: :index_testings_on_widget_type_and_widget_id)
+    assert connection.index_exists?(:testings, [:gizmo_type, :gizmo_id], name: :index_testings_on_gizmo_type_and_gizmo_id)
+  end
+end
+
+module LegacyPolymorphicReferenceIndexTest
+  ["6.0", "5.2", "5.1", "5.0", "4.2"].each do |version|
+    klass = Class.new(ActiveRecord::TestCase)
+    klass.class_eval <<~RUBY
+      include LegacyPolymorphicReferenceIndexTestCases
+
+      self.use_transactional_tests = false
+
+      def self.name
+        "V#{version.underscore}"
```

Comment:
Is underscore necessary for versions like "6.0"?

Reply:
It is only used to give anonymous classes in tests a name for easier debugging.
```
Error:
#test_polymorphic_add_reference_uses_all_column_names_in_index
```

vs
```
Error:
V6.0#test_polymorphic_add_reference_uses_all_column_names_in_index:
```

Reply:
Sorry for the ate reply, but I don't think `underscore` does anything for the values in:
```["6.0", "5.2", "5.1", "5.0", "4.2"]```.

Reply:
Ah, right, I misinterpreted it as like "why we need these versions in name".